### PR TITLE
Expose a single sql-scratchpad component

### PR DESCRIPTION
**Pull request by JohanAhlen**
One component containing everything needed to edit, execute and view the results.

Usage:

```
import sqlScratchpadComp from 'gethue/lib/components/SqlScratchpadWebComponent';

sqlScratchpadComp.setBaseUrl('http://banana');
await sqlScratchpadComp.login('hue', 'hue');
```

in html

`<sql-scratchpad dialect="mysql" />`

### CODE REVIEW

File: desktop/core/src/desktop/js/api/utils.ts [hunk 1]
```diff
@@ -53,6 +54,36 @@ export interface ApiFetchOptions<T, E = string> extends AxiosRequestConfig {
...
+
+let baseUrl = (window as hueWindow).HUE_BASE_URL;
+let bearerToken: string | undefined;
+
+axiosInstance.interceptors.request.use(config => {
+  if (baseUrl) {
+    config.baseURL = baseUrl;
+  }
+  if (bearerToken) {
+    config.headers['Authorization'] = `Bearer ${bearerToken}`;
```
Comment (romainr) [inline desktop/core/src/desktop/js/api/utils.ts:67]: Bearer --> JWT ? (tried with compose API? only supported there AFAIK)

I will update the UI there so use the compo, will be easy to test!

**Review by romainr — APPROVED** [review]
Uber magic!

File: desktop/core/src/desktop/js/api/utils.ts (hunk 1 above)
Comment (JohanAhlen) [reply desktop/core/src/desktop/js/api/utils.ts:67]: Bearer is the standard, https://jwt.io/introduction <- search for "bearer"

Comment (romainr) [reply desktop/core/src/desktop/js/api/utils.ts:67]: Yes maybe, but won't work with our libs, JWT is what we use and not wrong in itself

Comment (JohanAhlen) [reply desktop/core/src/desktop/js/api/utils.ts:67]: Probably for old reasons, you can configure with JWT_AUTH_HEADER_PREFIX': 'Bearer'

Comment (romainr) [reply desktop/core/src/desktop/js/api/utils.ts:67]: Indeed, perfecto, gracias!

**Pull request merged**